vignettes/slopes.Rmd

```diff
@@ -15,103 +14,390 @@ knitr::opts_chunk$set(
+
+```{r, eval=FALSE}
+usethis::edit_r_environ()
+MAPBOX_API_KEY=xxxxx # replace XXX with your api key
+```
+
+
+## Functions 
+
+### Elevation
+
+- `elevation_add()` Take a linestring and add a third dimension (z) to its coordinates
+- `elevation_get()` Get elevation data from hosted maptile services
+- `elevation_extract()` Extract elevations from coordinates
+
+- `z_value()` retrieves elevation values for each z (as vector of sequential vertices)
+- `z_start()` retrieves the elevation value of the first linestring vertice
+- `z_end()` retrieves the elevation value of the last linestring vertice
+- `z_mean()` retrieves the elevation mean value 
+- `z_max()`  retrieves the elevation max value 
+- `z_min()`  retrieves the elevation min value 
+
+### Distance
+
+- `sequential_dist()` Calculate the sequential distances between sequential coordinate pairs
+
+### Slope
 
-$$
-s = \Delta z/l
-$$
+- `slope_vector()` calculates the slopes associated with consecutive elements in one dimensional distance and associated elevations.
+- `slope_distance()` calculates the slopes associated with consecutive distances and elevations.
+- `slope_distance_mean()` calculates the mean average slopes associated with consecutive distances and elevations.
+- `slope_distance_weighted()` calculates the slopes associated with consecutive distances and elevations, with the mean value associated with each set of distance/elevation vectors weighted in proportion to the distance between each elevation measurement, so longer sections have proportionally more influence on the resulting gradient estimate.
 
-Depending on the purpose of application, it might me relevant to understand how hilliness is estimated.  
+- `slope_raster()` Calculate slopes of linestrings based on local raster map
+- `slope_matrix()` Calculate the gradient of line segments from a 3D matrix of coordinates
+- `slope_matrix_weighted()` Calculate the weighted gradient of line segments from a 3D matrix of coordinates
 
-# Implementation in the slopes package
+- `slope_xyz()` Calculates the slope associated with linestrings that have xyz coordinates
 
-Using the `slopes` package we can estimate the gradient of individual road segments. When these segments are combined into routes, we then need a means of assessing the hilliness of the entire route. A range of indices can be used to represent route hilliness. The choice of which index is most appropriate may be context dependent.
+### Plot
 
-To help research slopes using open source software for reproducibility, we have developed the slopes package.
-After you have installed it, load it as follows:
+- `plot_dz()` Plot a digital elevation profile based on xyz data
+- `plot_slope()` Plots the slope profile associated with a linestring with base R graphics
 
-```{r setup}
+## Package datasets
```

Comment:
This is really useful. We could also add links to the docs on the website but we can sort that out after.